### ISSUE TITLE
false negative unused_mut:  if let x .. = &mut

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust#69701]
````rust
fn main() {
   let mut word = "hello".to_string();
   if let Some(_) = &mut word.chars().last() { // here
       word.push_str(" word");
   }
   
   println!("{}", word);
}
````

In this example, the `mut` in `if let Some(char) = &mut word.chars().last() {` is redundant but the compiler issued no warning.

`rustc 1.43.0-nightly (4ad624882 2020-03-03)`

**Comment by jonas-schievink** [comment]
`unused_mut` only warns for local variables AFAIK

**Comment by cuviper** [comment]
Does `unused_mut` _ever_ trigger for `&mut`? I thought that was only for `mut` bindings.

```rust
fn main() {
    let x = &mut 42;
    println!("{}", *x);
}
```

No warning there...

**Comment by jeremystucki** [comment]
I’ll try to implement this

**Comment by jeremystucki** [comment]
Okay it is way harder than I thought, so I won't be able to get it to work
